internal/dryrun/transport.go

```diff
@@ -37,24 +30,13 @@ func (t *DryRunTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	case http.MethodTrace:
 	case http.MethodHead:
 	default:
-		obj, ok := runtimeObjectFrom(req.Context())
-		if !ok {
-			return nil, errors.New("no object present in context: cannot record dry run without an object")
-		}
-
-		meta, ok := obj.(metav1.ObjectMetaAccessor)
-		if !ok {
-			return nil, errors.New("object does not implement ObjectMetaAccessor")
-		}
-
 		verb, ok := verbMap[req.Method]
 		if !ok {
 			verb = "execute " + req.Method
 		}
 		msg := "Would %v %v"
-		t.Recorder.Eventf(obj, v1.EventTypeNormal, DryRunReason, msg, verb, req.URL.Path)
 
-		return nil, NewDryRunError(obj.GetObjectKind(), meta, v1.EventTypeNormal, DryRunReason, msg, verb, req.URL.Path)
+		return nil, NewDryRunError(nil, nil, "", "", msg, verb, req.URL.Path)
```

Comment:
nit: feels like way to many parameters. Consider using builder. Not critical

Reply:
see https://github.com/mongodb/mongodb-atlas-kubernetes/pull/2085#discussion_r1939463080, this is going away.